src/NonLinearSolver.f90

```diff
@@ -0,0 +1,429 @@
+    CALL this%func%clear()
+    DEALLOCATE(this%func)
+  ENDIF
+  this%func => NULL()
+  this%isInit=.FALSE.
+
+ENDSUBROUTINE clear_NonLinearSolverBase
+!
+!-------------------------------------------------------------------------------
+!> @brief Defines the interface for initializing a @c NonLinearSolver_Native object
+!> @param this the object to initialize
+!> @param ce the computing environment
+!> @param f the @c NonLinearSolverInterface_Base object to use for function evals
+!> @param plist the parameter list to use for initialization
+!>
+!> This procedure takes ownership of @c f and clears the input pointer.  It does
+!> not take ownership of @c ce.
+!>
+SUBROUTINE init_NonLinearSolverNative(this,ce,f,plist)
+  CLASS(NonLinearSolver_Native),INTENT(INOUT) :: this
+  TYPE(FutilityComputingEnvironment),TARGET,INTENT(IN) :: ce
+  CLASS(NonLinearSolverInterface_Base),INTENT(INOUT),POINTER :: f
+  TYPE(ParamType),INTENT(IN) :: plist
+  !
+  CHARACTER(LEN=*),PARAMETER :: myName='init_NonLinearSolverNative'
+  TYPE(ParamType) :: linSysPlist
+
+  REQUIRE(.NOT.this%isInit)
+  REQUIRE(ASSOCIATED(f))
+
+  CALL this%initBase(ce,f,plist)
+
+  IF(this%solverMethod == NLSOLVER_METHOD_BISECTION) THEN
+    CALL ce%exceptHandler%raiseFatalError(modName//'::'//myName//' - Bisection method '// &
+        'is not yet supported by the native NonLinearSolver object!')
+  ENDIF
+  this%TPLType=NLSOLVER_TPL_NATIVE
+
+  CALL linSysPlist%add('LinearSolverType->TPLType',NATIVE)
+  CALL linSysPlist%add('LinearSolverType->solverMethod',GE)
+  CALL linSysPlist%add('LinearSolverType->MPI_Comm_ID',PE_COMM_SELF)
+  CALL linSysPlist%add('LinearSolverType->numberOMP',1_SNK)
+  CALL linSysPlist%add('LinearSolverType->timerName','ODEtimer')
+  CALL linSysPlist%add('LinearSolverType->matType',DENSESQUARE)
+  CALL linSysPlist%add('LinearSolverType->A->MatrixType->n',this%n)
+  CALL linSysPlist%add('LinearSolverType->A->MatrixType->isSym',.FALSE.)
+  CALL linSysPlist%add('LinearSolverType->x->VectorType->n',this%n)
+  CALL linSysPlist%add('LinearSolverType->b->VectorType->n',this%n)
+  CALL this%linSys%init(linSysPlist)
+  CALL linSysPlist%clear()
+
+  this%isInit=.TRUE.
+
+ENDSUBROUTINE init_NonLinearSolverNative
+!
+!-------------------------------------------------------------------------------
+!> @brief Defines the interface for clearing a @c NonLinearSolver_Native object
+!> @param this the object to clear
+!>
+!> This routine clears data on the @c func component and then deallocates the
+!> pointer
+!>
+SUBROUTINE clear_NonLinearSolverNative(this)
+  CLASS(NonLinearSolver_Native),INTENT(INOUT) :: this
+
+  CALL this%clearBase()
+
+ENDSUBROUTINE clear_NonLinearSolverNative
+!
+!-------------------------------------------------------------------------------
+!> @brief Solves a @c NonLinearSolver_Native object
+!> @param this the @c NonLinearSolver_Native object to solve
+!> @param x the @c VectorType containing the starting guess for the solve
```

Comment:
x is also the final solution?

Reply:
I'll update the comment to reflect that.

Reply:
done